components/screens/deliveries/NewDelivery.tsx

```diff
@@ -99,38 +100,54 @@ class NewDelivery extends React.Component<Props, State> {
       defaultOpeningHours: [{
         opens: moment().startOf("day").toDate(),
         closes: moment().endOf("day").toDate()
-      }]
+      }],
+      deliveryQualities: []
     };
   }
 
   /**
    * Component did mount life-cycle event
    */
   public async componentDidMount() {
+    const { accessToken } = this.props;
     const { selectedDate } = this.state;
-    if (!this.props.accessToken) {
+
+    if (!accessToken) {
       return;
     }
 
     this.setState({ loading: true });
     const Api = new PakkasmarjaApi();
-    const productsService = await Api.getProductsService(this.props.accessToken.access_token);
-    const unfilteredProducts: Product[] = await productsService.listProducts(undefined, this.props.itemGroupCategory, this.props.accessToken.userId, undefined, 999);
+    const productsService = Api.getProductsService(accessToken.access_token);
+    const productPricesService = Api.getProductPricesService(accessToken.access_token);
+    const deliveryPlacesService = Api.getDeliveryPlacesService(accessToken.access_token);
+    const deliveryQualitiesService = Api.getDeliveryQualitiesService(accessToken.access_token);
+
+    const [ unfilteredProducts, deliveryPlaces ] = await Promise.all([
+      productsService.listProducts(undefined, this.props.itemGroupCategory, accessToken.userId, undefined, 999),
+      deliveryPlacesService.listDeliveryPlaces()
+    ]);
+
     const products = unfilteredProducts.filter(product => product.active === true);
-    const deliveryPlacesService = await Api.getDeliveryPlacesService(this.props.accessToken.access_token);
-    const deliveryPlaces = await deliveryPlacesService.listDeliveryPlaces();
-    const productPricesService = await Api.getProductPricesService(this.props.accessToken.access_token);
-    const productPrice: ProductPrice[] = products[0] ? await productPricesService.listProductPrices(products[0].id || "", "CREATED_AT_DESC", undefined, undefined, 1) : [];
     const deliveries = this.getDeliveries();
+    
+    const [ productPrice, deliveryQualities ] = await Promise.all([
+      products[0] ? await productPricesService.listProductPrices(products[0].id || "", "CREATED_AT_DESC", undefined, undefined, 1) : [],
+      (await deliveryQualitiesService.listDeliveryQualities(ItemGroupCategory.FRESH, products[0].id || "")).reverse()
```

Comment:
sort instead by descending priceBonus amount

Reply:
Now sorted so that highest bonus will be displayed first